src/cmd/run.js

```diff
@@ -42,16 +44,28 @@ export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir,
     onSourceChange = defaultSourceWatcher,
+    desktopNotifications = notifier,
   }: WatcherCreatorParams
  ): Watchpack {
   return onSourceChange({
     sourceDir,
     artifactsDir,
     onChange: () => {
       log.debug(`Reloading add-on ID ${addonId}`);
+
       return client.reloadAddon(addonId)
+        .then(() => {
+          desktopNotifications({
```

Comment:
Hi all, just catching up with this.

I really love the idea of seeing a notification on error but showing a notification every time it reloads seems a bit much to me. I think it would minimize the effect of seeing a notification for failures (i.e. you might start to ignore the notices).

Do you really think we need to show a notice on every reload? If you do, I would request not to make this the default behavior but to allow it with a new option: `--notify-on-reload`. I can see how reload notifications might be useful to trust that the system is running.